backend/db/resolve_fetch_failures_0_members.py

```diff
@@ -35,71 +36,104 @@ def _report_success(
     if not failed_cset_ids:
         print("No failures to resolve.")
         return
 
     # Resolve
     i = 0
     t0 = datetime.now()
+    cset_is_draft_map: Dict[int, bool] = {}
     print(f"Fetching concept set versions and their related objects: {', '.join([str(x) for x in failed_cset_ids])}")
-    while len(failed_cset_ids) > 0 and (datetime.now() - t0).total_seconds() < 2 * 60 * 60:  # 2 hours
+    while len(failed_cset_ids) > 0 and (datetime.now() - t0).total_seconds() < runtime:
         i += 1
         print(f"- attempt {i}: fetching members for {len(failed_cset_ids)} concept set versions")
+        # Check for new failures: that may have occurred during runtime
+        failed_cset_ids, failure_lookup_i = get_failures_0_members(version_id, use_local_db)
+        failure_lookup.update(failure_lookup_i)
         # Fetch data
         csets_and_members: Dict[str, List[Dict]] = fetch_cset_and_member_objects(
-            codeset_ids=failed_cset_ids, handle_issues=False)
+            codeset_ids=list(failed_cset_ids), flag_issues=False)
         # - filter success & track results
         csets_and_members['OMOPConceptSet'] = [
             x for x in csets_and_members['OMOPConceptSet'] if x['member_items']]
+        cset_is_draft_map.update({x['properties']['codesetId']: x['properties']['isDraft'] for x in csets_and_members['OMOPConceptSet']})
         success_cases: List[int] = [
             x['properties']['codesetId'] for x in csets_and_members['OMOPConceptSet']]
         failed_cset_ids = list(set(failed_cset_ids) - set(success_cases))
 
-        # Update DB
+        # Update DB & report success
         if success_cases:
             try:
                 with get_db_connection(schema=schema, local=use_local_db) as con:
                     concept_set_members__from_csets_and_members_to_db(con, csets_and_members)
                     refresh_derived_tables(con)
+                print(f"Successfully fetched concept set members for concept set versions: "
+                      f"{', '.join([str(x) for x in success_cases])}")
+                _report_success(success_cases, failure_lookup, 'Success result: Found members', use_local_db)
             except Exception as err:
                 reset_temp_refresh_tables(schema)
                 raise err
 
-        # Report success
-        if success_cases:
-            print(f"Successfully fetched concept set members for concept set versions: "
-                  f"{', '.join([str(x) for x in success_cases])}")
-            _report_success(success_cases, failure_lookup, 'Success result: Found members', use_local_db)
-
-        # Sleep
+        # Sleep or exit
+        if not loop:
+            break
         time.sleep(polling_interval_seconds)
 
+    # Remove drafts from failures (as csets don't have members until drafts are finalized)
+    failed_cset_ids = [cset_id for cset_id in failed_cset_ids if cset_is_draft_map[cset_id]]
```

Comment:
## Don't error on drafts
Action now doesn't fail if cset is still a draft when the refresh completes.